### PR TITLE
Workaround for the coloring order issue.

### PR DESCRIPTION
**Pull request by ps-pat**
Fix the issues #153 and #163. I don't know why that works but it seems to do. Kinda look like there is some deeper issue there, if anyone have any pointer please let me know!

### CODE REVIEW

**Comment by JackDevine** [comment]
Bisection shows that this came up in #146 

I was trying to fix an issue where Plots.jl started to treat NaN separated lists differently. I obviously didn't test corner cases enough. If Plots.jl can restore the old behaviour, then we could revert #146 and we would not have this issue. Alternatively, we will have to work out what went wrong in #146 

Interesting that shifting the node names by -1 fixes the issue for the graph in #163 

I tested with `g = ones(9, 9)` and it seems that the circshift fix doesn't work.

I think that the trick will be to look at #146 and try to work out why it is producing a list of vectors in the wrong order on line 924.

Basically, the logic of that part of the code is that it is trying to give the plotting recipe data for the perimeters of the nodes. In the past, we used to use NaN to indicate that the plotter should move to the next node. Now we have each node outline as it's own vector in a vector of vectors (we actually have a 2-tuple where the x coordinates are in the first element of the tuple and the y coordinates are in the second).

**Comment by ps-pat** [comment]
I didn't test my fix enough apparently! The weird thing is that the bug doesn't seem to occur with other attributes like `nodeshape` for instance. I played around with #146 and the list of vectors produced by your fix is in the same order than the one produced by older code. Maybe this is a Plot.jl issue?

**Comment by JackDevine** [comment]
> I didn't test my fix enough apparently!

That makes two of us!

> Maybe this is a Plot.jl issue?

Interesting theory. The whole motivation for #146 was to deal with the new way that Plots.jl deals with `Shape`s. If we downgrading Plots.jl fixes the problem, then that would prove your point.

Right now, it is quite difficult to disentangle the `graphplot` code from the Plots.jl code. I wonder if it is possible to come up with a really simple example where colors and labels are handled differently by Plots.jl?

**Comment by BeastyBlacksmith** [comment]
This might relate to https://github.com/JuliaPlots/StatsPlots.jl/issues/442
